app/src/main/java/com/washingtondcsquad/tudee/presentation/screens/add_task/AddNewTaskScreen.kt

```diff
@@ -0,0 +1,233 @@
+package com.washingtondcsquad.tudee.presentation.screens.add_task
+
+import com.washingtondcsquad.tudee.presentation.components.DatePickerModal
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import com.washingtondcsquad.tudee.R
+import com.washingtondcsquad.tudee.presentation.components.AppTextField
+import com.washingtondcsquad.tudee.presentation.components.TaskPriorityCard
+import com.washingtondcsquad.tudee.presentation.design.AppTheme
+import com.washingtondcsquad.tudee.presentation.design.textStyle.defaultTextStyle
+import org.koin.androidx.compose.koinViewModel
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.material3.rememberModalBottomSheetState
+import java.time.format.DateTimeFormatter
+import androidx.compose.foundation.lazy.items
+import androidx.compose.ui.graphics.Color
+import com.washingtondcsquad.tudee.presentation.components.CancelableActionLayout
+import com.washingtondcsquad.tudee.presentation.components.CategoryCard
+import java.time.LocalDate
+import java.time.ZoneId
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AddNewTaskScreen(
+    viewModel: AddTaskViewModel = koinViewModel(),
+    onCancelAddTaskBottomSheet: () -> Unit,
+    taskDate: LocalDate = LocalDate.now()
+) {
+
+    val uiState by viewModel.uiState.collectAsState()
+
+    val dateAsMilliseconds = taskDate
+            .atStartOfDay(ZoneId.systemDefault())
+            .toInstant()
+            .toEpochMilli()
+
+    viewModel.onDateSelected(dateAsMilliseconds)
```

Comment:
this is creates a side effect when this screen get recomposed this line will be executed